modules/source/sftp/config/sftp.xml

```diff
@@ -21,7 +21,8 @@
 		<beans:property name="password" value="${password}" />
 		<beans:property name="privateKey" value="${privateKey}" />
 		<beans:property name="privateKeyPassphrase" value="${passPhrase}" />
-		<beans:property name="allowUnknownKeys" value="true"/>
+		<beans:property name="allowUnknownKeys" value="${allowUnknownKeys}" />
+		<beans:property name="knownHosts" value="#{'UNDEFINED'.equals('${knownHostsExpression}') ? null : ${knownHostsExpression}}" />
```

Comment:
That would work too - I went this route because `${knownHostsExpression}` didn't work with a null property (unresolved property).

That seems wrong to me - it prevents explicitly setting a property to `null`, although I guess the DSL doesn't support that anyway.

I was also thinking of the generated docs - default: `UNDEFINED` seems more doc friendly than default: null.

But I can change it if you prefer.